### PR TITLE
Remove `Strategy.optimizer_zero_grad`

### PR DESCRIPTION
**Pull request by daniellepintz**
## What does this PR do?
Remove `Strategy.optimizer_zero_grad` since no Strategy subclasses this hook, and we can just call the LM hook directly

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
I disagree with the removal from Strategy.
The strategy can intercept the optimizer.step() so it should also be able to do that with the zero_grad() call imo.

File: CHANGELOG.md [hunk 1]
```diff
@@ -346,6 +346,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Removed `Strategy.init_optimizers` in favor of `Strategy.setup_optimizers` ([#11236](https://github.com/PyTorchLightning/pytorch-lightning/pull/11236))
 
+
+- Removed `Strategy.optimizer_zero_grad` ([#11246](https://github.com/PyTorchLightning/pytorch-lightning/pull/11246))
```
Comment (Borda) [inline CHANGELOG.md:350]:
```suggestion
- Removed `Strategy.optimizer_zero_grad` for duplicite functionality ([#11246](https://github.com/PyTorchLightning/pytorch-lightning/pull/11246))
```

**Review by Borda — APPROVED** [review]
